recipes/skyr-url/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.28.0"
+
+class SkyrUrlConan(ConanFile):
+    name = "skyr-url"
+    homepage = "https://cpp-netlib.github.io/url"
+    description = "A C++ library that implements the WhatWG URL specification"
+    topics = ("conan", "whatwg", "url", "parser")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSL-1.0"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_json": [True, False], "with_fs": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_json": True, "with_fs": True}
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package_multi"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder (self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        # https://github.com/cpp-netlib/url#requirements
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++17 support. The current compiler {} {} does not support it.".format(
+                    self.name, self.settings.compiler, self.settings.compiler.version))
+                
+        if self.options.with_fs and str(self.settings.compiler) == "apple-clang":
+            raise ConanInvalidConfiguration("apple-clang 11 is currently not support with filesystem")
+            
+        if self.options.shared:
```

Comment:
```suggestion
        if self.options.shared:
```

shared could be removed from options, but okay as it raises a ConanInvalidConfiguration

Reply:
I'm pretty sure it's possible and someone else may want it in the future, I just don't have the time to figure it out right now. Something is better than nothing

Reply:
yes, it makes sense.